runtime/src/block_min_prioritization_fee_cache.rs

```diff
@@ -0,0 +1,595 @@
+    }
+
+    fn report_metrics_for_slot(&mut self, slot: Slot) {
+        // report current min fees for slot
+        if let Some(block) = self.get_block_min_prioritization_fee(&slot) {
+            datapoint_info!(
+                "block_min_prioritization_fee",
+                ("slot", slot as i64, i64),
+                ("entity", "block", String),
+                (
+                    "min_prioritization_fee",
+                    block.get_block_fee().unwrap_or(0) as i64,
+                    i64
+                ),
+            );
+            for (account_key, fee) in block.get_account_fees() {
+                datapoint_info!(
+                    "block_min_prioritization_fee",
+                    ("slot", slot as i64, i64),
+                    ("entity", account_key.to_string(), String),
+                    ("min_prioritization_fee", *fee as i64, i64),
+                );
+            }
+        }
+        // report and reset counter
+        {
+            datapoint_info!(
+                "block_min_prioritization_fee_counters",
+                ("slot", slot as i64, i64),
+                (
+                    "successful_transaction_update_count",
+                    self.successful_transaction_update_count as i64,
+                    i64
+                ),
+            );
+            datapoint_info!(
+                "block_min_prioritization_fee_counters",
+                ("slot", slot as i64, i64),
+                (
+                    "fail_get_transaction_priority_details_count",
+                    self.fail_get_transaction_priority_details_count as i64,
+                    i64
+                ),
+            );
+            datapoint_info!(
+                "block_min_prioritization_fee_counters",
+                ("slot", slot as i64, i64),
+                (
+                    "fail_get_transaction_account_locks_count",
+                    self.fail_get_transaction_account_locks_count as i64,
+                    i64
+                ),
+            );
+            datapoint_info!(
+                "block_min_prioritization_fee_counters",
+                ("slot", slot as i64, i64),
+                (
+                    "fail_finalize_block_not_found",
+                    self.fail_finalize_block_not_found as i64,
+                    i64
+                ),
+            );
+            self.successful_transaction_update_count = 0;
+            self.fail_get_transaction_priority_details_count = 0;
+            self.fail_get_transaction_account_locks_count = 0;
+            self.fail_finalize_block_not_found = 0;
+        }
+    }
+}
+
+impl BlockMinPrioritizationFeeCacheUpdate for BlockMinPrioritizationFeeCache {
+    /// Update block's min prioritization fee with `txs`,
+    /// Returns updated min prioritization fee for `slot`
+    fn update_transactions<'a>(
+        &mut self,
+        slot: Slot,
+        txs: impl Iterator<Item = &'a SanitizedTransaction>,
+    ) -> Option<u64> {
```

Comment:
Did you say you are expecting to use this return value in the banking_stage path?
If not, I don't think it's worth supporting this just for tests; we can inspect the cache directly

Reply:
It is in back of my mind to reuse the `PrioritizationFee` code in TPU, to share cache between sigverify stage (read-only) and banking stages (write), so transactions with lesser than current min fee can be dropped pre-sigverify. But I haven't get down to details impl yet. It may not use the return after all, i am happy to remove it makes code cleaner.